lib/zclient.h

```diff
@@ -792,6 +804,22 @@ struct zclient_options {
 	bool synchronous;
 };
 
+/* link layer representation for GRE like interfaces
+ * pfx_in is the underlay IP, pfx_out is the tunnel dest
+ * index stands for the index of the interface
+ * ndm state stands for the NDM value in netlink
+ */
+#define ZEBRA_NEIGH_STATE_REACHABLE (0x02)
+#define ZEBRA_NEIGH_STATE_FAILED    (0x20)
+struct zapi_neigh_ip {
+	int cmd;
+	struct ipaddr ip_in;
+	struct ipaddr ip_out;
+	ifindex_t index;
+	uint32_t ndm_state;
+};
+int zclient_neigh_ip_get(struct stream *s, struct zapi_neigh_ip *api);
```

Comment:
please call this "_decode", and add a corresponding "_encode" function for use in nhrp: the nhrp code is still encoding with raw stream api calls.